Intersect.Client/Core/Graphics.cs

```diff
@@ -151,51 +151,74 @@ public static void DrawIntro()
 
         public static void DrawMenu()
         {
-            // Draw an animated background in the main menu.
-            if (ClientConfiguration.Instance.MenuBackgroundAnimated)
+            // Animated background in the main menu.
+            if (ClientConfiguration.Instance.MenuBackground.Count > 1)
             {
-                DrawFullScreenTextureAnimated(
-                    ClientConfiguration.Instance.MenuBackground,
-                    ClientConfiguration.Instance.MenuBackgroundFrameInterval,
-                    ClientConfiguration.Instance.MenuBackgroundDisplayMode
+                sMenuBackground = sContentManager.GetTexture(
+                    TextureType.Gui, ClientConfiguration.Instance.MenuBackground[sMenuBackgroundIndex]
                 );
+
+                if (sMenuBackground == null)
+                {
+                    return;
+                }
+
+                if (sMenuBackgroundInterval < Timing.Global.Milliseconds)
+                {
+                    sMenuBackgroundIndex++;
+                    sMenuBackgroundInterval = Timing.Global.Milliseconds +
+                                              ClientConfiguration.Instance.MenuBackgroundFrameInterval;
+                }
+
+                if (sMenuBackgroundIndex >= ClientConfiguration.Instance.MenuBackground.Count)
+                {
+                    sMenuBackgroundIndex = 0;
```

Comment:
I think `sMenuBackgroundIndex = (sMenuBackgroundIndex + 1) % ClientConfiguration.Instance.MenuBackground.Count;` is better ( you can combine both if statements).